plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -76,3 +76,32 @@ fu_plugin_init_vfuncs(FuPluginVfuncs *vfuncs)
 	vfuncs->device_registered = fu_plugin_thunderbolt_device_registered;
 	vfuncs->device_created = fu_plugin_thunderbolt_device_created;
 }
+
+gboolean
+fu_plugin_composite_prepare (FuPlugin *plugin,
```

Comment:
On main the way this works, you need to declare the vfuncs for composite prepare and cleanup as `static` and then declare what they are in `fu_plugin_init_vfuncs`.  You can look at `plugins/dell-dock/fu-plugin-dell-dock.c` for an example of how this is supposed to be done.